lib/engine_socketio.js

```diff
@@ -118,20 +118,21 @@ SocketIoEngine.prototype.step = function (requestSpec, ee) {
     return engineUtil.createLoopWithCount(requestSpec.count || -1, steps);
   }
 
-  let f = function(context, callback, socketio) {
+  let f = function(context, callback) {
     // Only process emit requests; delegate the rest to the HTTP engine (or think utility)
     if (requestSpec.think) {
-      return engineUtil.createThink(requestSpec);
+      return engineUtil.createThink(requestSpec, _.get(self.config, 'defaults.think', {}));
     }
     if (!requestSpec.emit) {
       let delegateFunc = self.httpDelegate.step(requestSpec, ee);
       return delegateFunc(context, callback);
     }
     ee.emit('request');
     let startedAt = process.hrtime();
+    let socketio = context.sockets[requestSpec.emit.namespace] || null;
 
-    if (!(requestSpec.emit && requestSpec.emit.channel)) {
-      ee.emit('error', 'invalid arguments');
+    if (!(requestSpec.emit && requestSpec.emit.channel && socketio)) {
+      return ee.emit('error', 'invalid arguments');
```

Comment:
@hassy, I added socketio to be required here either. And I return after the emit. Continuing execution without these properties seems unnecessary / problematic.